src/lib/redux-cozy-client/slices/sharings.js

```diff
@@ -105,24 +117,29 @@ export const fetchSharings = (doctype, id = null, options = {}) => ({
   promise: (client) => client.fetchSharings(doctype)
 })
 
-export const share = (document, recipient, sharingType, sharingDesc) => async (dispatch, getState) => {
-  const recipientId = recipient.id || (await dispatch(createContact(recipient))).data[0].id
+export const share = (document, recipients, sharingType, sharingDesc) => async (dispatch, getState) => {
+  const recipientIds = await Promise.all(
+    recipients.map(recipient => recipient.id || dispatch(createContact(recipient)).then(c => c.data[0].id))
+  )
   trackSharingByEmail(document)
-  return dispatch(createSharing(document, recipientId, sharingType, sharingDesc))
+  return dispatch(createSharing(document, recipientIds, sharingType, sharingDesc))
 }
 
 export const unshare = (document, recipient) => async (dispatch, getState) => {
   const sharing = getSharingForRecipient(getState(), document, recipient)
+  const loneRecipient = sharing.attributes.recipients.length === 1
   return dispatch({
     types: [REVOKE_SHARING, RECEIVE_SHARING_REVOKE, RECEIVE_ERROR],
     doctype: document._type,
     id: document._id,
     sharingId: sharing.attributes.sharing_id,
-    // TODO: right now, we create one sharing for each recipient, so we can just
-    // delete the sharing, but when we'll have many recipients for one sharing,
-    // we'll need to use another route:
-    // https://github.com/cozy/cozy-stack/blob/master/docs/sharing.md#delete-sharingssharing-idrecipientclient-id
-    promise: client => client.revokeSharing(sharing.attributes.sharing_id)
+    recipientId: recipient._id,
+    promise: client => loneRecipient
+      ? client.revokeSharing(sharing.attributes.sharing_id)
+      : client.revokeSharingForClient(
+        sharing.attributes.sharing_id,
+        sharing.attributes.recipients.find(r => r.recipient.id === recipient._id).Client.client_id
```

Comment:
What is this `Client` property with a capital _«C»_?

Reply:
I'm not responsible for properties naming on the server side </passive-agressive>